src/zimscraperlib/zim/indexing_item.py

```diff
@@ -0,0 +1,199 @@
+from zimscraperlib.zim.items import StaticItem
+
+
+class IndexData(libzim.writer.IndexData):
+    """IndexData to properly pass indexing title and content to the libzim
+
+    Both title and content have to be customized (title can be identical to item title
+    or not).
+    keywords is optional since it can be empty
+    wordcount is optional ; if not passed, it is automaticaly computed from content
+    """
+
+    def __init__(
+        self, title: str, content: str, keywords: str = "", wordcount: int | None = None
+    ):
+        self.title = title
+        self.content = content
+        self.keywords = keywords
+        self.wordcount = wordcount
+
+    def has_indexdata(self) -> bool:
+        return len(self.content) > 0 or len(self.title) > 0
+
+    def get_title(self) -> str:
+        return self.title
+
+    def get_content(self) -> str:
+        return self.content
+
+    def get_keywords(self) -> str:
+        return self.keywords
+
+    def get_wordcount(self) -> int:
+        return (
+            self.wordcount
+            if self.wordcount is not None
+            else len(self.content.split()) if self.content else 0
+        )
+
+
+class IndexingItem(StaticItem):
+    """A StaticItem capable to customize indexing data (title and content).
+
+    When indexdata (with title and content) is passed, this is used for indexing.
+
+    Otherwise, indexdata is automatically computed based on item content when possible.
+
+    For now only PDF documents are supported for automatic computation of index data,
+    and search index is populated only with PDF text content (i.e. no OCR is done on PDF
+    images)
+    """
+
+    def __init__(
+        self,
+        content: str | bytes | None = None,
+        fileobj: io.BytesIO | None = None,
+        filepath: pathlib.Path | None = None,
+        path: str | None = None,
+        title: str | None = None,
+        mimetype: str | None = None,
+        hints: dict | None = None,
+        indexdata: IndexData | None = None,
+        **kwargs: Any,
+    ):
+
+        super().__init__(
+            content, fileobj, filepath, path, title, mimetype, hints, **kwargs
+        )
+
+        if indexdata:
+            # indexdata has been externaly configured
+            self.get_indexdata = lambda: indexdata
+        else:
+            self.get_auto_title_and_index_data()
+
+    def get_auto_title_and_index_data(self):
+        """Populate item index data and title automatically from content"""
+
+        # content was set manually
+        content = getattr(self, "content", None)
+        if content is not None:
+            if not isinstance(content, (str, bytes)):
+                raise AttributeError(
+                    f"Unexpected type for content: {type(content)}"
+                )  # pragma: no cover
+            mimetype = get_content_mimetype(
+                content.encode() if isinstance(content, str) else content
```

Comment:
Is this a ruff-recommendation? I find explicit `encode(UTF8)` more practical.

Reply:
Nope it is not, fixed.